src/value.cc

```diff
@@ -614,7 +624,7 @@ gpointer V8ToGHash (GITypeInfo *type_info, Local<Value> value) {
 
 item_error:
         /* Free everything we have converted so far. */
-        FreeGIArgument(type_info, (GIArgument *) &hash_table, GI_TRANSFER_NOTHING, GI_DIRECTION_IN);
+        FreeGIArgument(type_info, (GIArgument *) &hash_table, transfer, GI_DIRECTION_IN);
```

Comment:
Oh I think this one is wrong, isn't it? Here we're handling the error case and we want all of this to be freed.

Reply:
Indeed, you're right. I've reverted this part, and add a comment to clarify this to prevent similar mistake at this location.